payjoin/src/receive/mod.rs

```diff
@@ -672,7 +672,11 @@ pub struct ProvisionalProposal {
 impl ProvisionalProposal {
     /// Apply additional fee contribution now that the receiver has contributed input
     /// this is kind of a "build_proposal" step before we sign and finalize and extract
-    fn apply_fee(&mut self, min_feerate: Option<FeeRate>) -> Result<&Psbt, RequestError> {
+    fn apply_fee(
+        &mut self,
+        min_feerate: Option<FeeRate>,
+        max_feerate: Option<FeeRate>,
```

Comment:
`min_feerate` is optional because it has a floor, but `max_feerate` prevents an attack. I'm not so sure it should be optional. I suppose a min_feerate transaction could get stuck, but bitcoin core lets you double spend spent coins after a restart to fix the attack, or CPFP, where a max_feerate attack could lose funds with no recourse.

Reply:
Agreed.